contracts/LpSugar.vy

```diff
@@ -673,8 +702,15 @@ def _positions(
           positions.append(pos)
 
     else:
+      for alm_position_id in alm_positions_ids:
+        alm_position: AlmManagedPositionInfo = alm_core.managedPositionAt(alm_position_id)
+        if IPool(alm_position.pool).factory() == factory.address:
+          for ammPositionId in alm_position.ammPositionIds:
+            alm_positions_nfpm_ids.append(ammPositionId)
```

Comment:
This needs confirmation

Reply:
Here you also should take a share of user position, because pool position totally owned by regarding `LpWrapper`, so there is necessary to handle `LpWrapper.balanceOf(user)/LpWrapper.totalSupply()` coupled to `nfpm_id` to the next calculation of token amounts and etc.

Please see code example: https://github.com/mellow-finance/mellow-alm-toolkit/blob/aerodrome/src/scripts/deploy/base/DepositWithdraw.s.sol